content/get-haiku/r1beta4/release-notes.md

```diff
@@ -0,0 +1,189 @@
+
+This was originally done to port GTK3, and so it may be missing some (or many) API calls needed by other applications, though it has been tested with a wide variety of X11 applications and most non-Motif ones seem to at least partially work already.
+
+### Wayland compatibility layer
+
+Yes, in addition to an X11 compatibility layer, we now also have one for Wayland! It is a little more complicated than the one for X11, running an "in-process Wayland server" for each application instead of translating C API calls directly. The GTK3 port now is built atop this instead of the X11 compatibility layer for both features and performance reasons.
+
+## Changes &amp; bug fixes
+
+Hundreds of bugs and other minor issues were resolved since the previous release, and there are simply too many to try and list them all here. Virtually every area of the system received some amount of attention, and the result is Haiku's most polished and stable release yet, even if some rough edges remain here and there.
+
+### Improved POSIX compatibility
+
+Haiku is always looking to see how we can improve our adherence to POSIX and make it easier for developers to port their software. There are too many changes in this area to individually list, but here are some highlights:
+
+ * Fixed behavior of `O_NOCTTY`, which GPG relies on.
+ * Addition and implementation of `locale_t` and related methods.
+ * Support for C11 threads.
+ * Increased maximum argument length to match FreeBSD (roughly doubled).
+ * ELF "undefined weak symbols" now supported.
+ * `dl_iterate_phdr` implemented.
+
+### Boot failure fixes
+
+Boot failures have wide and diverse causes, from USB driver bugs to race conditions in initialization and back again. Many problems preventing successful boots completely or intermittently both on real hardware and in virtual machines were tracked down and resolved, making it possible to at least start Haiku on a much wider variety of hardware than before.
+
+### NVMe driver improvements
+
+The NVMe driver now supports falling back to "polling" mode when interrupts are not working properly, handles "unaligned" transfers much more reliably, and had a number of performance improvements made to it. Support for TRIM was also implemented.
+
+### Kernel always compiled with newer GCC
+
+On 32-bit x86 systems, we still use the ancient GCC 2.95 to compile may parts of the system to maintain ABI compatibility with BeOS. However, in this past development cycle, changes were made so that the kernel and drivers are always compiled with the newer GCC version (presently GCC 11), even when building most of the system with GCC 2.95.
```

Comment:
```suggestion
On 32-bit x86 systems, we still use the ancient GCC 2.95 to compile many parts of the system to maintain ABI compatibility with BeOS. However, in this past development cycle, changes were made so that the kernel and drivers are always compiled with the newer GCC version (presently GCC 11), even when building most of the system with GCC 2.95.
```